skbio/metadata/_intersection.pyx

```diff
@@ -186,7 +186,15 @@ cdef class IntervalNode:
         if self.cleft is not EmptyNode and self.cleft.maxend > start:
             self.cleft._intersect( start, end, results )
         # This interval
-        if ( self.end > start ) and ( self.start < end ):
+        if self.end == self.start:
+            send = self.end
+        else:
+            send = self.end - 1
```

Comment:
The original code was `self.end > start`, but when that changed to `self.end >= start` you needed to subtract one from `self.end` to match the new check (when start != end). Or at least I think that is why this code is written that way...